docs/vm-configuration.md

```diff
@@ -264,3 +264,95 @@ the generation of multiple libvirt XML documents, the domain XML and associated
 documents for secrets, network filters and node devices. So there is not a strict
 1-1 mapping between the k8s API resource representation and the libvirt low
 level representation.
+
+Proposed VM Definition
+----------------------
+
+The following definition adheres to main points described above. It provides
+
+1. high level description usable by end users or management applications and
+2. low level details required to construct and launch a VM.
+
+The features section is used to generate the domain section, the other way
+around is out of scope for this proposal. Each feature must represent one or
+more fields in the domain section.
+
+The design and contents of each feature section should be logically split and
+discussed in separate proposal. This is just an example without the sections
+discussed, converted from cluster/vm.yaml. Splitting the features enables us to
+create modular VM Spec generation flows that are easily tunable by experts in
+their domains of expertise.
+
+```yaml
+apiVersion: kubevirt.io/v1alpha2
+kind: VM
+metadata:
+  name: testvm
+spec:
+  features:
```

Comment:
If we really say, that the `feature` section allows a full description, then it is probably not so attractive to have the feature section on the `VM` object too. Then I would just use `VMConfig` and store the VMSpec for VM restarts on the VMConfig. Even though reusing as much as possible in the context of admission controllers would be something I love I would then try to avoid the conflict situations right now. If we would also allow on the `VMConfig` to set values in `feature` and `domain`, then I would  tend to having something like these `features` on the `VM`, since we already have to deal with  conflicts anyway, and the runtime could profit from it. E.g. if we introduce something like a `VirtualMachineReplicationController`, it could still work with the `VM`, and would not have to go through `VMConfig`, or have a `VirtualMachineReplicationControllerConfig` to provide it's features.